asmcomp/linscan.ml

```diff
@@ -41,6 +53,15 @@ let rec insert_interval_sorted i = function
   | j :: _ as il when j.iend <= i.iend -> i :: il
   | j :: il -> j :: insert_interval_sorted i il
 
+let rec release_expired_spilled ci pos =
```

Comment:
All other `release_expired_*` functions are return a value that must be subsituted for a field of `ci` (sometimes mutating `ci` as well), but this one is imperative and only mutates `ci`.
It would maybe be possible to clean those other functions to be consistent as well, and combine the code of `walk_interval` into a single `release_expired` function mutating `ci`. I am however not sure whether such a cleanup belong to a pull request like this or not.

Reply:
Indeed, the asymmetry is jarring. I went ahead and pushed a commit uniformizing the style of the four functions by doing the mutation of `ci` inside each one. More extensive refactoring could be done later.